service/access.go

```diff
@@ -0,0 +1,395 @@
+/*
+Copyright (C)  2018 Yahoo Japan Corporation Athenz team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package service
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kpango/fastime"
+	"github.com/kpango/gache"
+	"github.com/kpango/glg"
+	ntokend "github.com/kpango/ntokend"
+	"github.com/pkg/errors"
+	"github.com/yahoojapan/athenz-client-sidecar/config"
+	"golang.org/x/sync/singleflight"
+)
+
+// AccessService represent a interface to automatically refresh the access token, and a access token provider function pointer.
+type AccessService interface {
+	StartAccessUpdater(context.Context) <-chan error
+	RefreshAccessTokenCache(ctx context.Context) <-chan error
+	GetAccessProvider() AccessProvider
```

Comment:
```suggestion
	GetAccessTokenProvider() AccessTokenProvider
```

Reply:
same as above.

Reply:
#36

In this PR, proceed with current name convention.